includes/class-wp-security-bp-database.php

```diff
@@ -142,15 +151,21 @@ public function check_name() {
 		$args['data']       = $this->db_name;
 
 		//push domain to $db_names_blacklist (coming soon)
-		$domain = $this->wpdb->get_var('SELECT option_value FROM wp_options WHERE option_name="home"');
-		array_push(
-			$this->db_names_blacklist,
-			preg_replace( '#^http(s?)?:\/\/|(w{3}\.)?(\.[a-z]{2,3})?#', '', $domain )
+		$domains = preg_replace(
+			'#^http(s?)?:\/\/|(w{3}\.)?(\.[a-z]{2,3})?|-|_#',
+			'',
+			array(
+				$this->domain_site,
+				$this->domain_home,
+			)
 		);
 
 		$check = ! in_array(
 			preg_replace( '#\.|_|-#', '', $this->db_name ),
-			$this->db_names_blacklist
+			array_merge(
+				$this->db_names_blacklist,
+				$domains
+			)

```

Comment:
PHPCS suggests using strict comparison for `in_array()`. This is done by setting the third argument to `true`. However I find it not really necessary here.

Reply:
Ok. Man. We all use PHPCS.
before leaving here this comment... Could you ask me? 
;)

Reply:
Why not using strict comparison?

Reply:
Why not using strict comparison?

Reply:
sorry, too early at the morning. the comment was not polite at all. I see it now.

The point here is, Do we need to comment sniffer warnings here, if they're not relevant for the result of the code? Let's discuss this one, so. I think this shouldn't, never, stop a PR (I know the stopper is the bad regex, which I find so clever and well find :+1: ). The amount of possible reasons to keep or don't see a warning is big. From my point of view, let's avoid stop the merge end we'll fix it in next PRs (leaving a comment in slack or an issue if you want)

here, we have to agree. If we don't need the strict mode, why to use it. And you answer "why not to use it". So, this is an infinite discussion, unless we find or define a law for this situation. If it's a better practice, as codeSniffer says (wich make sense), ok move on with strict mode. But does it really is? Do we have to consider something that makes absolutely nothing here, to be in place just to avoid a warning?... for me the answer was "not". But I want to be open-minded with this.

Reply:
https://stackoverflow.com/questions/20248666/in-this-case-which-is-better-in-matching-regular-expression-or-parse-url

https://bugs.php.net/bug.php?id=16545

according to this. performance of `parse_url` is not so good. As I improved the regex I think could be good to keep it. What do you think?

Reply:
>  I think this shouldn't, never, stop a PR

Totally agree, we are just developing the first release so a request changes should only be used if the PR breakes the code. Other changes could just be marked with a "to do" message. Sorry about that.

> Do we have to consider something that makes absolutely nothing here, to be in place just to avoid a warning?

Honestly, I'm surprised about your point here :open_mouth:. I remember you recently saying that `==` comparison operator should not exist. Using `in_array()` in loose mode is the same as using `==`!! It's not to just to shut up the warnings but to follow the best practices. Otherwise, why still using a comma for the last results in arrays? :man_shrugging: 

I think we should always follow the WPCS unless there is a good reason for not doing it.

Reply:
ok! I'll set it to strict.

`==` this killed me. nice shoot.